src/components/swap/SwapPlatformSelector.tsx

```diff
@@ -155,54 +159,58 @@ export function SwapPlatformSelector({
+              <SelectionListOption
+                role="button"
+                key={trade.platform.name}
                 isSelected={selectedTrade?.platform.name === trade.platform.name}
+                onClick={() => handleSelectedTradeOverride(trade.platform.name)}
               >
-                <StyledFlex alignItems="center">
-                  {ROUTABLE_PLATFORM_LOGO[trade.platform.name]}
-                  {trade.platform.name}
-                </StyledFlex>
-              </SelectionListName>
-              <SelectionListDetails>
-                <PriceImpact priceImpact={priceImpactWithoutFee} />
-              </SelectionListDetails>
-              <SelectionListDetails>{realizedLPFee ? `${realizedLPFee.toFixed(2)}%` : '-'}</SelectionListDetails>
-              {showGasFees && (
+                <SelectionListName
+                  flex={showGasFees ? '30%' : '45%'}
+                  isSelected={selectedTrade?.platform.name === trade.platform.name}
+                >
+                  <StyledFlex alignItems="center">
+                    {ROUTABLE_PLATFORM_LOGO[trade.platform.name]}
+                    {trade.platform.name}
+                  </StyledFlex>
+                </SelectionListName>
                 <SelectionListDetails>
-                  <GasFee loading={debouncedLoadingGasFees} gasFeeUSD={gasFeeUSD} />
+                  <PriceImpact priceImpact={priceImpactWithoutFee} />
                 </SelectionListDetails>
-              )}
-              <SelectionListReceiveAmount flex="25%">
-                <TYPE.subHeader color="white" fontSize="12px" fontWeight="600">
-                  {tokenAmount === '0' ? '<0.0000001' : tokenAmount || '-'}
-                </TYPE.subHeader>
-                <CurrencyLogo
-                  currency={isExactIn ? trade.outputAmount.currency : trade.inputAmount.currency}
-                  size="14px"
-                  marginLeft={4}
-                />
-              </SelectionListReceiveAmount>
-            </SelectionListOption>
-          )
-        })}
+                <SelectionListDetails>{realizedLPFee ? `${realizedLPFee.toFixed(2)}%` : '-'}</SelectionListDetails>
+                {showGasFees && (
+                  <SelectionListDetails>
+                    <GasFee loading={debouncedLoadingGasFees} gasFeeUSD={gasFeeUSD} />
+                  </SelectionListDetails>
+                )}
+                <SelectionListReceiveAmount flex="25%">
+                  <TYPE.subHeader color="white" fontSize="12px" fontWeight="600">
+                    {tokenAmount === '0' ? '<0.0000001' : tokenAmount || '-'}
+                  </TYPE.subHeader>
+                  <CurrencyLogo
+                    currency={isExactIn ? trade.outputAmount.currency : trade.inputAmount.currency}
+                    size="14px"
+                    marginLeft={4}
+                  />
+                </SelectionListReceiveAmount>
+              </SelectionListOption>
+            )
```

Comment:
As I'm looking into this code, I'm thinking about turning this into a new component to improve readability. It is just a suggestion, we could make an issue and solve it latter if you think it should belong in another PR/task.

Reply:
I have the same concern about readability here. I created a ticket: #1015, and we can do that next.